src/Avalonia.Base/Rendering/RenderLoop.cs

```diff
@@ -97,7 +98,14 @@ private void TimerTick(TimeSpan time)
                 {
                     bool needsUpdate = false;
 
-                    foreach (IRenderLoopTask item in _items)
+                    lock (_items)
+                    {
+                        _itemsCopy.Clear();
+                        foreach (var i in _items)
+                            _itemsCopy.Add(i);
```

Comment:
Why not set the Capacity before adding items?
```suggestion
                        _itemsCopy.Clear();
                        _itemsCopy.Capacity = _items.Count;
                        foreach (var i in _items)
                            _itemsCopy.Add(i);
```